packages/shadergradient-v2/src/WithFramerControls.tsx

```diff
@@ -1,10 +1,41 @@
-import { ShaderGradient as OriginalShaderGradient } from './ShaderGradient'
-import { ControlType } from 'framer'
-import { ShaderGradientWithControls } from './types'
+import { ShaderGradient } from './ShaderGradient'
+import { addPropertyControls, ControlType } from 'framer'
+import { GradientT } from './types'
 
-const ShaderGradient = OriginalShaderGradient as ShaderGradientWithControls
+type FramerShaderGradientProps = GradientT & {
+  position?: {
+    positionX: number
+    positionY: number
+    positionZ: number
+  }
+  rotation?: {
+    rotationX: number
+    rotationY: number
+    rotationZ: number
+  }
+}
+
+function FramerShaderGradient({
+  position,
+  rotation,
+  ...rest
+}: FramerShaderGradientProps) {
+  const { positionX, positionY, positionZ } = position
+  const { rotationX, rotationY, rotationZ } = rotation
+  return (
+    <ShaderGradient
+      positionX={positionX}
+      positionY={positionY}
+      positionZ={positionZ}
+      rotationX={rotationX}
+      rotationY={rotationY}
+      rotationZ={rotationZ}
+      {...rest}
+    />
+  )
+}
```

Comment:
The `FramerShaderGradient` component receives position and rotation objects and decomposes them into positionX, positionY, positionZ, and rotationX, rotationY, rotationZ properties, which it then passes to the Framer.